lib/google_search_data_viewer/accounts/user.ex

```diff
@@ -0,0 +1,40 @@
+defmodule GoogleSearchDataViewer.Accounts.User do
+  use Ecto.Schema
+
+  import Ecto.Changeset
+
+  alias GoogleSearchDataViewer.Accounts.Passwords
+
+  schema "users" do
+    field :email, :string
+    field :password, :string, virtual: true
+    field :hashed_password, :string
+
+    timestamps()
+  end
+
+  def changeset(user, attrs) do
+    user
+    |> cast(attrs, [:email, :password])
+    |> validate_required([:email, :password])
+  end
+
+  def register_changeset(user, attrs) do
+    user
+    |> cast(attrs, [:email, :password])
+    |> validate_required([:email, :password])
+    |> validate_format(:email, ~r/^[^\s]+@[^\s]+$/, message: "must have the @ sign and no spaces")
+    |> validate_length(:password, min: 12)
+    |> unsafe_validate_unique(:email, GoogleSearchDataViewer.Repo)
```

Comment:
Why do we need this extra check? I think `unique_constraint` is enough for checking the uniqueness of the email 💭

Reply:
True, I think if I was to use this it should be used early on and not at the end - ie it provides no early feedback to the user prior to other validations. I think I initially added it when the constraint wasn't enforced when I forgot to add the actual constraint to the database, removed.